### PR TITLE
Bonding allocator

### PR DESCRIPTION
**Pull request by JExtor**

### CODE REVIEW

File: contracts/allocators/BondingAllocator.sol [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    function bond(
+        address _depository, 
+        address _token, 
+        uint256 _amount, 
+        uint256 _maxPrice, 
+        bool _managing
+    ) external onlyGuardian {
+        if(_managing) {
+            // retrieve amount of token from treasury
+            treasury.manage(_token, _amount);
```
Comment (lolchocotaco) [inline contracts/allocators/BondingAllocator.sol:64]: Don't understand this `if` block. In what scenario would we want managing == false?

**Review by lolchocotaco — COMMENTED** [review]
Overall looks okay. 
Could you add a little more context around which depositories we'd be bonding to? 

Is this essentially olympus treasury, to OP bonds?

File: contracts/allocators/BondingAllocator.sol (hunk 1 above)
Comment (JExtor) [reply contracts/allocators/BondingAllocator.sol]: Was thinking if there were tokens sent to the contract directly to be bonded. After second thought probably not needed and doesn't matter. Removed.

**Comment by lolchocotaco** [comment]
Hey, closed the branch this PR was based off of, can you make it of of Main?

**Comment by JExtor** [comment]
@lolchocotaco Off of main now

File: contracts/allocators/BondingAllocator.sol [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+pragma solidity ^0.8.10;
+
+import "../libraries/SafeERC20.sol";
+
+import "../interfaces/IERC20.sol";
+import "../interfaces/ITreasury.sol";
+
+import "../types/OlympusAccessControlled.sol";
+
+interface IBond {
```
Comment (PARABRAHMAN0) [inline contracts/allocators/BondingAllocator.sol:11]: Test

**Pull request closed without merging**